client/state/sites/selectors/has-business-or-ecommerce-plan.js

```diff
@@ -0,0 +1,16 @@
+/**
+ * External dependencies
+ */
+import { get } from 'lodash';
```

Comment:
Basically, I am not sure whether creating a `siteHasBusinessOrEcommercePlan` selector is the best way forward. Since themes & plugins can only be installed on Atomic, isn't `isAtomic` selector https://github.com/Automattic/wp-calypso/blob/2de5e432fc90d3793d8bac4f8aa3e7a7dfb5432c/client/state/selectors/is-site-automated-transfer.js#L13 a better option?

Reply:
>Since themes & plugins can only be installed on Atomic

You can also have a Free Atomic site, and in this case you should see the Calypso upgrade nudge which would not happen if I used the `isAtomic` selector?

Reply:
Ahh, yes! That's correct! In that case I am still wondering whether we need a selector or we can bake it directly in the component. Have you figured out we may reuse this?

Reply:
We also need the `isAtomic` on top, cause having a Business on WPCOM should still show the nudge (that will initiate the automatic transfer). Instead I got an error

![](https://cln.sh/duko3K+)

Reply:
I think the selector might be useful but we should rename it to something like `canUploadCustomThemesAndPlugins` which fundamentally checks the sites plan for this? This way we can use the selector for the same problem in the plugins section and if the ability for this feature expands beyond Business and Ecommerce plans we can just update the selector?

I am trying to shift the mindset away from Simple vs Atomic checks to plan checks.

Reply:
>We also need the isAtomic on top

I think we should keep this check outside the selector. Let me make some amends.